core/src/raw/path.rs

```diff
@@ -157,7 +157,7 @@ pub fn get_basename(path: &str) -> &str {
     if !path.ends_with('/') {
         return path
             .split('/')
-            .last()
+            .next_back()
```

Comment:
Nice catch! However, it's a bit harder to understand compared to `last()`. Would you consider adding a comment to explain it?

Reply:
of course will add it

Reply:
> I think I would rather go the other way around, warning about last when iterator implements DoubleEndedIterator. The problem with last is that it doesn't require DoubleEndedIterator to be implemented which means that in general case it's going to be slow, and while this can be dealt with manual implementation of that method, most iterators in standard library don't bother to do so. For those that do bother, I don't expect any particular difference between next_back and last even when the iterator is to be immediately dropped.


more: https://github.com/rust-lang/rust-clippy/issues/1822